docs/tutorials/python/table.md

```diff
@@ -1,2 +1,221 @@
+
+```python
+results = query(
+    f"SELECT * FROM {table.id} WHERE Chromosome = '1' AND Start < 41000 AND \"End\" > 20000"
+)
+for _, row_info in results.iterrows():
+    print(row_info)
+```
+
+
+## 3. Changing Data
+
+Once the schema is settled, changes come in two flavors: appending new rows and updating existing ones.
+
+### Appending new rows
+
+Appending new rows is fairly straightforward. To continue the previous example, we might add some new genes from another file.
+Lets take this new data in `more_genes.csv`
+
+```csv
+Name,Chromosome,Start,End,Strand,TranscriptionFactor
+Qux1,4,201001,202001,+,False
+Qux2,4,203001,204001,+,False
+```
+
+To add it via a filepath
+
+```python
+table = Table(
+    name="My Favorite Genes",
+    parent_id=project.id,
+).get()
+table = table.store_rows(values="/path/to/more_genes.csv")
+```
+
+Using pandas:
+
+```python
+new_rows_df = pd.DataFrame({
+    "Name": ["Qux3", "Qux4"],
+    "Chromosome": ["4", "4"],
+    "Start": [201001, 203001],
+    "End": [202001, 204001],
+    "Strand": ["+", "+"],
+    "TranscriptionFactor": [False, False]
+})
+table.store_rows(values=new_rows_df)
+```
+
+### Updating existing rows
+
+Updating rows requires an etag, which identifies the most recent change set plus row IDs and version numbers for each row to be modified. We get those by querying before updating. Minimizing changesets to contain only rows that actually change will make processing faster.
+
+For example, let's update the names of some of our favorite genes:
+
+```python
+results_df = query(f"select * from {table.id} where Chromosome='1'")
+results_df['Name'] = ['rzing', 'zing1', 'zing2', 'zing3']
+table.store_rows(values=results_df)
+```
+
+If your table has a primary key, you can use the upsert functionality to update existing rows or insert new rows based on whether the primary key value already exists in the table. This function does not do deletions. Here is an example.
+
+```python
+to_upsert_df = pd.DataFrame({
+    "Name": ["Qux3", "Qux5"],
+    "Chromosome": ["4", "4"],
+    "Start": [201001, 203001],
+    "End": [202001, 204001],
+    "Strand": ["-", "+"],
+    "TranscriptionFactor": [True, False]
+})
+# Qux3 will be updated, Qux5 will be inserted
+table.upsert_rows(values=to_upsert_df,  primary_keys=['Name'])
+```
+
+## 4. Changing Table Structure
+
+* Adding columns
+
+    ```python
+    table.add_column(
+        Column(name="Expression", column_type="STRING")
+    )
+    table.store()
+    ```
+
+* Renaming or modifying a column:
+
+    ```python
+    table.columns['Expression'].name = 'Expression2'
+    table.columns['Expression'].column_type = 'INTEGER'
+    table.store()
+    ```
+
+* Removing a column
+
+    ```python
+    table.delete_column(name="Expression2")
+    table.store()
+    ```
+
+## 5. Deleting rows
+
+Query for the rows you want to delete and call syn.delete on the results:
+
+```python
+table.delete_rows(query=f"SELECT * FROM {table.id} WHERE Strand = '+'")
```

Comment:
I think we will want to work on being able to delete a file based on a filtered dataframe.  Take for example:

1. do the table query
2. do complex filtering (not with SQL) to find cols to delete
3. table.delete_rows(values=df)